### DISCUSSION TITLE
Akka.DependencyInjection and Previous IWindsorInstaller Registrations

### DISCUSSION THREAD

**Discussion by marcoshass** [akkadotnet/akka.net#6098] [Q&A] [answered]
Hi, I am replacing **Akka.DI.Core** and **Akka.DI.CastleWindsor** by Akka.DependencyInjection in a project.
Several Actors have constructor injected dependencies and these were being registered by the DI.CastleWindsor library (inside of a class implementing the IWindsorInstaller interface) like below:

`
public class ActorInstaller : IWindsorInstaller
{
        public void Install(IWindsorContainer container, IConfigurationStore store)
        {
            // register the actors for their dependencies
       }
}
`

By removing **Akka.DI.Core** and **Akka.DI.CastleWindsor** I lost all previous IWindsorInstalller registrations since now I have to use Microsoft.Extensions.DependencyInjection to register the dependencies for the Actors like below:

` services.AddScoped<IDatabaseRepository>(k => return new DatabaseRepository()});`

Question: Is it possible to reuse the IWindsorInstaller registrations?

**Comment by Aaronontheweb** [comment, marked as answer]
I'm not experienced with Castle Windsor, but I know they maintain an adapter for working with Microsoft.Extensions.DependencyInjection here: https://www.nuget.org/packages/Castle.Windsor.Extensions.DependencyInjection/ - maybe that will let you re-use your old `IWindsorInstaller` instances?

**Reply by Aaronontheweb** [reply]
@marcoshass was this able to help you?

**Reply by marcoshass** [reply]
Thanks it worked!

**Reply by Aaronontheweb** [reply]
Glad to hear it!

**Comment by marcoshass** [comment]
Thanks Aaron it worked!

Em sáb., 17 de set. de 2022 às 14:35, Aaron Stannard <
***@***.***> escreveu:
